### PR TITLE
committing, might need to reformat time in response

### PR DESCRIPTION
**Pull request by Brums16**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -169,3 +169,93 @@ describe("GET /api/articles", () => {
...
+      .get("/api/articles/50/comments")
+      .then(({ text }) => {
+        expect(text).toBe("No article found for article_id 50");
+      });
+  });
+  test("responds with a 404 code when passed an article id for an article with no comments", () => {
+    return request(app).get("/api/articles/8/comments").expect(404);
+  });
+  test("responds with appropriate error message when passed a valid article id with no comments", () => {
+    return request(app)
```
Comment (Harrison365) [inline __tests__/app.test.js:258]: These don't have to be separate requests :) 

However, an article with no comments should still be a 200 but return an empty array.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -169,3 +169,93 @@ describe("GET /api/articles", () => {
...
+      .get("/api/articles/8/comments")
+      .then(({ text }) => {
+        expect(text).toBe("No comments found for article_id 8");
+      });
+  });
+});
```
Comment (Harrison365) [inline __tests__/app.test.js:264]: Make sure to do a test for an invalid comment id e.g. /api/articles/banana/comments

File: models/topics-model.js [hunk 3]
```diff
@@ -48,3 +48,26 @@ exports.fetchAllArticles = async () => {
...
+
+exports.fetchCommentsByArticleId = async (id) => {
+  const { rows } = await db.query(
+    `
+      SELECT articles.article_id, comments.author, comments.body, comment_id, comments.created_at, comments.votes FROM articles LEFT JOIN comments
+      ON articles.article_id = comments.article_id
+      WHERE articles.article_id = $1
+      ORDER BY created_at DESC
+     `,
+    [id]
```
Comment (Harrison365) [inline models/topics-model.js:60]: Clever SQL to avoid multiple database queries 👍 🥇

File: models/topics-model.js [hunk 4]
```diff
@@ -48,3 +48,26 @@ exports.fetchAllArticles = async () => {
...
+  if (!rows[0])
+    return Promise.reject({
+      status: 404,
+      msg: `No article found for article_id ${id}`,
+    });
+  else if (rows[0].comment_id === null)
+    return Promise.reject({
+      status: 404,
+      msg: `No comments found for article_id ${id}`,
+    });
```
Comment (Harrison365) [inline models/topics-model.js]: This shouldn't be an error. Should just return an empty array. :)

**Comment by Harrison365** [comment]
Hi Chris :) 
Almost there on this one:
- No comments doesn't mean an error.
- Please add a test for an invalid id e.g. banana

Thanks 😄

File: models/topics-model.js (hunk 4 above)
Comment (Brums16) [reply models/topics-model.js]: Was wondering about this one, that makes it easier to code anyway

File: __tests__/app.test.js (hunk 2 above)
Comment (Brums16) [reply __tests__/app.test.js:264]: Is it ok if this still returns the same error message?

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**